internal/expr/prepare.go

```diff
@@ -204,24 +204,6 @@ func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, []outputDest,
 	var info *info
 	var ok bool
 
-	for _, t := range p.target {
-		info, ok = ti[t.prefix]
-		if !ok {
-			return nil, nil, fmt.Errorf(`type %s unknown, have: %s`, t.prefix, strings.Join(getKeys(ti), ", "))
```

Comment:
If we remove this check it will also disappear for Case 1. We would need to add in a check there that the `Person` in `&Person.*` corresponded to a real type. Otherwise line 211 will case a panic.